bin/frequency_calculator.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * Calculate the importance (amount of stopping trains) for a station.
+ * Stations with a higher weight are more important.
+ *
+ * @Author Bertware
+ */
+
+// Constants
+const GTFS_ZIP = 'http://gtfs.irail.be/nmbs/nmbs-latest.zip';
+const TMP_UNZIP_PATH = 'nmbs-latest-gtfs';
+const TMP_ZIPFILE = 'nmbs-latest-gtfs.zip';
+const GTFS_STOP_TIMES = 'stop_times.txt';
+const STATIONS_CSV = '../stations.csv';
+
+/*
+ * Step 1 : Get the latest information from GTFS.
+ * This information can be found at http://gtfs.irail.be/nmbs/nmbs-latest.zip
+ */
+
+echo('Gathering resources...'.PHP_EOL);
+
+// Download zip file with GTFS data.
+file_put_contents(TMP_ZIPFILE, file_get_contents(GTFS_ZIP));
+
+// Load the zip file.
+$zip = new ZipArchive();
+if ($zip->open(TMP_ZIPFILE) != 'true') {
+    die('Could not extract downloaded GTFS data');
+}
+
+// Extract the zip file and remove it.
+$zip->extractTo(TMP_UNZIP_PATH);
+$zip->close();
+unlink(TMP_ZIPFILE);
+
+// Get the file we need.
+rename(TMP_UNZIP_PATH.'/'.GTFS_STOP_TIMES, GTFS_STOP_TIMES);
+
+echo('Cleaning up resources...'.PHP_EOL);
+// Remove temporary data.
+$tmpfiles = scandir(TMP_UNZIP_PATH);
+foreach ($tmpfiles as $file) {
+    if ($file != '.' && $file != '..') {
+        // Remove all extracted files from the zip file.
+    unlink(TMP_UNZIP_PATH.'/'.$file);
+    }
+}
+reset($tmpfiles);
+// Remove the empty folder.
+rmdir(TMP_UNZIP_PATH);
+
+/*
+ * Step 2: patch the csv file
+ * For this step, we need 3 actions:
+ * - Read the stop times, and create a frequency table for every stop id
```

Comment:
I'm afraid to say that some stop times can be way more important than others... For instance 1 stop times rule can be used by a gtfs:trip that is only used on Christmas and New Year, while another stop times rule is used every weekday.

While I do believe the autocomplete will work indeed, I think a better approach is also calculating from the calendar_dates.txt and calendar.txt how many times the stop time actually occurs